### ISSUE TITLE
Incorrect warnings

### ISSUE THREAD

**Issue by Idomo** [jonkykong/SideMenu#409]
## New Issue Checklist

I have read the [guidelines for contributing](https://github.com/jonkykong/SideMenu/blob/master/.github/CONTRIBUTING.md) and I understand:
- [ ] My issue is happening in the **latest version** of SideMenu.
- [x] My issue was **not** solved in the [README](https://github.com/jonkykong/SideMenu/blob/master/README.md).
- [x] My issue can **not** be answered on [stackoverflow.com](stackoverflow.com).
- [x] My issue is **not** a request for new functionality that I am unwilling to build and contribute with a pull request.
- [ ] My issue **is** reproducible in the [demo project](https://github.com/jonkykong/SideMenu/tree/master/Example).

## Issue Description

Hi,
Because my app supports RTL languages, I did something like this:
```
func getSideMenu(in vc: UIViewController) -> UISideMenuNavigationController {
  return isRTL(vc.view) ? SideMenuManager.default.menuRightNavigationController! : SideMenuManager.default.menuLeftNavigationController!
}
```
So sometimes I'm using the `menuLeftNavigationController` and sometimes the `menuRightNavigationController`.
I'v noticed that I'm getting warnings in the console that I used `menuAddScreenEdgePanGesturesToPresent` before I set the `menuLeftNavigationController`, because I'm using the `menuRightNavigationController`.
But, I also got `menuRightNavigationController cannot be modified while it's presented.` when I indeed used the `menuRightNavigationController` and I didn't changed anything in the `SideMenuManager.default` after doing it in the `viewDidLoad` of the `initial view controller`.

My `SideMenuManager.default` settings are:
```
/// Set the side menu settings
func initSideMenu(in vc: UIViewController){ // Called only in the initial view controller
  let menuNavigationController = vc.storyboard!.instantiateViewController(withIdentifier: "MenuNavigationController") as! UISideMenuNavigationController
  if isRTL(vc.view){
    SideMenuManager.default.menuRightNavigationController = menuNavigationController
  } else {
    SideMenuManager.default.menuLeftNavigationController = menuNavigationController
  }

  SideMenuManager.default.menuAddPanGestureToPresent(toView: vc.navigationController!.navigationBar)
  SideMenuManager.default.menuAddScreenEdgePanGesturesToPresent(toView: vc.navigationController!.view)
  SideMenuManager.default.menuFadeStatusBar = false // Prevent status bar from turning black
  SideMenuManager.default.menuAllowPushOfSameClassTwice = false // Prevent pushing same VC if it's already open
  SideMenuManager.default.menuPushStyle = .replace
  SideMenuManager.default.menuPresentMode = .viewSlideInOut
  SideMenuManager.default.menuAnimationFadeStrength = 0.5
  SideMenuManager.default.menuAnimationPresentDuration = 0.5
  SideMenuManager.default.menuAnimationDismissDuration = 0.5
  SideMenuManager.default.menuAnimationCompleteGestureDuration = 0.5
}
```

Can you help my to understand those warnings and fix them if I'm right about the right/left menu using?
It is noteworthy that everything works good; when using `present()` and when using gestures, I just didn't understood why I got those warnings and if I shouldn't get them, let you know what you have to fix. 

~BTW, I'm using version `~> 4.0.0` because I have other libraries that haven't upgraded to Swift 4.2 yet, so I can't update my code too.~

Thanks,
Ido.

**Comment by jonkykong** [comment]
If you look through the code you will see there are warnings when a menu item hasn't been set yet. This is mostly for beginner devs who make mistakes. You can set the environment variable `STFU_SIDEMENU` to `true` to silence those warnings.

You also can't change the menu while it's in the view hierarchy. Investigate your code further to make sure that you aren't changing the menus while they are still in the hierarchy (even if they aren't visible to the user).

**Comment by Idomo** [comment]
> If you look through the code you will see there are warnings when a menu item hasn't been set yet. This is mostly for beginner devs who make mistakes. You can set the environment variable `STFU_SIDEMENU` to `true` to silence those warnings.
> 
> You also can't change the menu while it's in the view hierarchy. Investigate your code further to make sure that you aren't changing the menus while they are still in the hierarchy (even if they aren't visible to the user).

Sorry for the late reply, I didn't have time to investigate this.
Now I checked and I really didn't understood the warning: 

> SideMenu Warning: menuAddScreenEdgePanGesturesToPresent was called before menuLeftNavigationController was set. The gesture will not work without a menu. Use menuAddScreenEdgePanGesturesToPresent(toView:forMenu:) to add gestures for only one menu.

I am using `menuAddScreenEdgePanGesturesToPresent(toView:forMenu:)` 'Cause I want to "add gestures for only one menu" (as it saying in the warning), but I didn't used `menuAddScreenEdgePanGesturesToPresent` as property (as much as I understood from this warning), so what does it want from me?
I'v also tried to set the `menuRightNavigationController`/`menuLeftNavigationController` to `nil` when I used the other one, but the warning is still presenting.

**Comment by jonkykong** [comment]
Read the code and you'll see the warnings are if either menu isn't set, and you're only using one instead of both. It's safe to ignore the warnings.

**Comment by Idomo** [comment]
But thats what I’m saying, I am using onaly one, but I’m using `menuAddScreenEdgePanGesturesToPresent(toView:forMenu:)` as rhe warning telling me to do when I want only one menu.
So why does the warning appears?
You can see my code in the first comment, it called only once in the initialViewController `viewDidLoad()` (I’v serounded the call with if that validate sideMenu is nill).
As you can see in the image, I did add this environment variable and I’m still getting the warnings, I didn’t asked if its a safe thing to do, I want to ignor them.

**Comment by jonkykong** [comment]
Warning is to prevent newbie developers from making mistakes and opening support tickets I have to explain. It’s more helpful for me to have gratuitous logging.

**Comment by Idomo** [comment]
Sorry, I'v missed the `forMenu:` part, maybe because of the time...
But I think it was nicer if you would explain the problem with the same breath you'r saying to look at the warnings, just for future comments.

**Comment by Idomo** [comment]
And also, note that the environment variable `STFU_SIDEMENU` doesn't hiding the warnings, and you can see in the image I'v attached that it is set properly.
